modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/CompositeArtifactExtensionManager.java

```diff
@@ -0,0 +1,130 @@
+    extensionModels.addAll(childExtensionManager.getExtensions());
+  }
+
+  @Override
+  public void registerExtension(ExtensionModel extensionModel) {
+    throw new UnsupportedOperationException("Composite extension manager cannot register extensions");
+  }
+
+  @Override
+  public Set<ExtensionModel> getExtensions() {
+    return extensionModels;
+  }
+
+  @Override
+  public Optional<ExtensionModel> getExtension(String extensionName) {
+    return extensionModels.stream().filter(extensionModel -> extensionModel.getName().equals(extensionName)).findFirst();
+  }
+
+  @Override
+  public ConfigurationInstance getConfiguration(String configurationProviderName, Event event) {
+    return getConfigurationProvider(configurationProviderName).map(provider -> provider.get(event))
+        .orElseThrow(() -> new IllegalArgumentException(
+                                                        format(
+                                                               "There is no registered configurationProvider under name '%s'",
+                                                               configurationProviderName)));
+  }
+
+  @Override
+  public ConfigurationInstance getConfiguration(ExtensionModel extensionModel, Event event) {
+
+    Optional<ConfigurationProvider> provider = getConfigurationProvider(extensionModel);
+    if (provider.isPresent()) {
+      return provider.get().get(event);
+    }
+
+    throw new IllegalArgumentException(format(
+                                              "There is no registered configuration provider for extension '%s'",
+                                              extensionModel.getName()));
+  }
+
+  @Override
+  public Optional<ConfigurationProvider> getConfigurationProvider(ExtensionModel extensionModel) {
+    Optional<ConfigurationProvider> configurationProvider = childExtensionManager.getConfigurationProvider(extensionModel);
+
+    if (!configurationProvider.isPresent()) {
+      configurationProvider = parentExtensionManager.getConfigurationProvider(extensionModel);
+    }
+
+    return configurationProvider;
+  }
+
+  @Override
+  public Optional<ConfigurationProvider> getConfigurationProvider(String configurationProviderName) {
+    Optional<ConfigurationProvider> configurationProvider =
+        childExtensionManager.getConfigurationProvider(configurationProviderName);
+
+    if (!configurationProvider.isPresent()) {
+      configurationProvider = parentExtensionManager.getConfigurationProvider(configurationProviderName);
+    }
+
+    return configurationProvider;
+  }
+
+  @Override
+  public void registerConfigurationProvider(ConfigurationProvider configurationProvider) {
+    throw new UnsupportedOperationException("Composite extension manager cannot register extension providers");
+  }
+
+  @Override
+  public ExtensionManifest parseExtensionManifestXml(URL manifestUrl) {
+    throw new UnsupportedOperationException("Composite extension manager cannot parse extension manifests");
```

Comment:
I can't provide a scenario, but if there's no impediment to fulfill this API I'd rather do it. The scenario might appear in the future and this becomes a bug we might not catch before hand. Liskov